toolchain/driver/compile_subcommand.cpp

```diff
@@ -420,6 +420,8 @@ class CompilationUnit {
 
   auto PreCheck() -> Parse::NodeLocConverter& {
     CARBON_CHECK(parse_tree_, "Must call RunParse first");
+    CARBON_CHECK(!node_converter_.has_value(), "Called PreCheck twice");
```

Comment:
In other calls (`RunLex`, `RunParse`, `PostCheck`, etc) we only verify that we're moving forward. They would share this problem. I'm fine with adding more checks to help document the progressive call structure, but can you please add them in the other spots too?

Reply:
Right, these functions do not `emplace` any optionals, so it's not clear how to CHECK that without adding a bool for the purpose of checking, or maybe an enum tracking the progress through them all if it's linear. WDYT?

Reply:
Note, we typically don't write `.has_value()` for optionals. `parse_tree_` is an `optional`. You could use the same CHECKs we're using to ensure things are called in order, to make sure they're not called again.

Reply:
Oh, thanks, will do. The lack of has_value() and use of operator= had me miss that. There's a couple functions that are still missed, but this covers most of them (and all the ones that set optionals).